zerver/tests/test_realm_export.py

```diff
@@ -0,0 +1,62 @@
+import mock
+
+from django.utils.timezone import now as timezone_now
+from django.conf import settings
+
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.test_helpers import (
+    use_s3_backend,
+    create_s3_buckets,
+)
+
+from zerver.views.public_export import public_only_realm_export
+from zerver.models import RealmAuditLog
+
+class RealmExportTest(ZulipTestCase):
+    def test_hit_api_endpoint(self) -> None:
+        admin = self.example_user('iago')
+        admin_email = admin.email
+        self.login(admin_email)
+        result = self.client_post('/json/exports/data_export')
+        self.assert_json_success(result)
+        self.logout()
+
```

Comment:
Finally, this test is incomplete.  We checked the endpoint succeeded, but not that event was queued or the export actually happened (and I think it will have, because the default model for our queue processors in the test suite is to immediately run the `consume` function for the queue processor when receiving an event, so that you can test the whole flow in a single unit test).  

We possibly don't want this to actually run the full export, but I think it makes sense to mock e.g. `do_export_realm` with `mock.patch` and verify the arguments passed into that using `m.call_args_list` (grep for it to see examples) so that we're testing all the code outside the core export logic here.